### ISSUE TITLE
Hapi: export hapiRequestInfoExtractor

### ISSUE THREAD

**Issue by AdriVanHoudt** [googleapis/nodejs-error-reporting#197]
(this is a feature request so I don't think the regular questions would help)

We log error is hapi sometimes with an error tag
`request.log(['user', 'create', 'error'], new Error('failed creating user'))`

To capture this now I do
```js
const HapiRequestExtractor = require('@google-cloud/error-reporting/build/src/request-extractors/hapi.js');

server.on('request', (request, event, tags) => {

    if (tags.error) {
        const requestInfo = HapiRequestExtractor.hapiRequestInformationExtractor(request);
        requestInfo.url = request.path; // see https://github.com/googleapis/nodejs-error-reporting/issues/196
        ErrorReporter.report(event.data || event.error, requestInfo);
    }
});
```

While I don't mind setting up custom logging it would be nice to not have to hack in the `hapiRequestInformationExtractor`

Another solution would be to support this use case out of the box.
Whichever you like I can do a PR for any/both.

**Comment by DominicKramer** [comment]
Thanks for opening this issue.  Just do double check, you are using Hapi 17 right?  Also, if issue #196 is resolved, the suggestion of this issue is to use `event.data || event.error` instead of just `event.error` (which is what the plugin currently uses).  This is correct?  Thanks

**Comment by AdriVanHoudt** [comment]
I'm using hapi 16, sorry for not mentioning that 😅 
The suggestion is to log logs logged with the error tag to stackdriver error reporting.
#196 will just make this logging more correct.
Atm doing `request.log(['user', 'create', 'error'], new Error('failed creating user'))` in hapi 16 does not log it to stackdriver afaik.

**Comment by ofrobots** [comment]
@AdriVanHoudt is this still needed now that #196 is fixed? I'm not a hapi user, so I don't know – without researching – how hapi error tags work.

**Comment by DominicKramer** [comment]
With #196 fixed, I'm going to close this for now.  If this issue is not resolved via the fix for #196 please let us know.
